frontend/packages/topology/src/components/side-bar/providers/TabSection.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { GraphElement, isEdge, observer } from '@patternfly/react-topology';
+import { useTranslation } from 'react-i18next';
+import { DetailsTab, DetailsTabSection, ResolvedExtension } from '@console/dynamic-plugin-sdk';
+import { Tab } from '@console/internal/components/utils';
+import { orderExtensionBasedOnInsertBeforeAndAfter } from '@console/shared';
+import { getResource } from '@console/topology/src/utils';
+import { DefaultResourceSideBar } from '../DefaultResourceSideBar';
+import TopologyEdgeResourcesPanel from '../TopologyEdgeResourcesPanel';
+
+type TabSection = Omit<
+  DetailsTabSection['properties'] & { resolvedSection: React.ReactNode },
+  'section' | 'tab'
+>;
+
+type TabSectionProps = {
+  element: GraphElement;
+  callback: (tabs: Tab[], loaded: boolean) => React.ReactElement;
```

Comment:
I'm not a fan (nor a strict opponent) of passing functions as children, but because SideBarTabLoader defines this callback as children, and its here also "a children" like callback, I would suggest to go both time with callback, or both times with children.

```suggestion
  children: (tabs: Tab[], loaded: boolean) => React.ReactElement;
```